### ISSUE TITLE
False positives for unused_import when referencing only static members of the imported module

### ISSUE THREAD

**Issue by mildm8nnered** [realm/SwiftLint#4199]
### New Issue Checklist

- [x ] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

I am getting false positives for `unused_import` in files that only references static members of an imported module.

For example, given

```
public class LogManager: NSObject {
    public class func log(_ message: String) {
        print(message)
    }
}
```

and the following in my main module:

```
import Cocoa
import Logging

@main
class AppDelegate: NSObject, NSApplicationDelegate {

    @IBOutlet var window: NSWindow!

    func applicationDidFinishLaunching(_ aNotification: Notification) {
        LogManager.log("hello")
    }
}
```

both `import Cocoa` and `import Logging` would get flagged up.

In fact, this simple example actually works fine, but in my much more complex multi-module project, which I can't share here, every `import Logging` is getting flagged up, and similarly for other modules

It looks like other people have also seen this issue:

https://stackoverflow.com/questions/72934586/swiftlint-analyzer-removes-imports-if-using-of-static-calls-not-desired-behavio

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint analyze
```

### Environment

* SwiftLint version 0.47.1
* Installation method used - CocoaPods, building from source, etc)?
* Paste your configuration file:

```yml
# only run on these files
included:
  - TestUnusedImports

excluded:
  - Pods/

analyzer_rules:
  - unused_import
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No
* Which Xcode version are you using (check `xcodebuild -version`)? 13.4.1 (13F100)
* Do you have a sample that shows the issue? No

**Comment by mildm8nnered** [comment]
This is no longer happening for me, with later versions of SwiftLint
